concert/devices/cameras/pco.py

```diff
@@ -1,112 +1,46 @@
 """PCO cameras implementation."""
 
-import asyncio
-import time
 from datetime import datetime
 import numpy as np
-from concert.coroutines.base import background
-from concert.devices.cameras import base
-from concert.devices.cameras.uca import Camera
-from concert.quantities import q
+import logging
+from concert.devices.cameras.uca import Camera as UcaCamera
 
 
-class Pco(Camera):
+LOG = logging.getLogger(__name__)
 
-    """Pco camera implemented by libuca."""
 
-    async def __ainit__(self):
-        await super().__ainit__('pco')
+class Camera(UcaCamera):
+    async def __ainit__(self, name="pco", params=None):
+        await super().__ainit__(name=name, params=params)
 
-    async def stream(self):
-        """
-        stream()
+    async def grab(self, index=None):
+        image = await self._grab_real(index=index)
+        if await self.get_timestamp_mode() in ["both", "binary"]:
```

Comment:
I think this should be cached? Especially with `net`, this means we every single time ask the server to ask the camera to get the timestamp mode, let's cache this.

Reply:
Done in the `start_readout` and `_start_real`.